From 4baea4670abcfb004f0d9fe1de654d12de1c7390 Mon Sep 17 00:00:00 2001
From: Jason Montleon <jmontleo@redhat.com>
Date: Wed, 12 Feb 2025 09:05:18 -0500
Subject: [PATCH 369/370] add missing include

---
 include/uapi/linux/dma_memcp.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/uapi/linux/dma_memcp.h b/include/uapi/linux/dma_memcp.h
index 3cb75a348ba8..94f3c16cbcfc 100644
--- a/include/uapi/linux/dma_memcp.h
+++ b/include/uapi/linux/dma_memcp.h
@@ -21,6 +21,8 @@
  *          Yuyang Cong <congyuyang@eswincomputing.com>
  */
 
+#include <stddef.h>
+
 #ifndef ES_DMA_MEMCP_H
 #define ES_DMA_MEMCP_H
 
-- 
2.47.0

