From 922d8de53339e2a99a42c5a1487528d532d4a44c Mon Sep 17 00:00:00 2001
From: yangwei1 <yangwei1@eswincomputing.com>
Date: Wed, 2 Apr 2025 13:45:41 +0800
Subject: [PATCH 458/551] fix:fix mpq8785 enable error and read voltage error

Changelogs:

Signed-off-by: yangwei1 <yangwei1@eswincomputing.com>
---
 drivers/regulator/mpq8785.c | 125 +++++++++++++++++++++++++-----------
 1 file changed, 89 insertions(+), 36 deletions(-)

diff --git a/drivers/regulator/mpq8785.c b/drivers/regulator/mpq8785.c
index 2e4e95e3d98b..a2bfd04af41b 100644
--- a/drivers/regulator/mpq8785.c
+++ b/drivers/regulator/mpq8785.c
@@ -192,7 +192,7 @@ struct MPQ8785_DRIVER_DATA
 static u32 garr_volt_numerator[] = {64, 80, 80, 80};
 static char garr_bool_string[][2] = {"N", "Y"};
 static char garr_speed_string[][8] = {"100kHz", "400kHz", "1MHz", "resv"};
-
+static u32 mpq8785_get_vout(struct MPQ8785_DRIVER_DATA *data);
 static struct of_regulator_match mpq8785_matches[] = {
 	{
 		.name = "npu_svcc",
@@ -238,8 +238,17 @@ static inline s32 mpq8785_str2ul(const char *buf, u32 *value)
 static u8 mpq8785_read_byte(struct MPQ8785_DRIVER_DATA *data, u8 command)
 {
 	int ret = 0;
+	int cnt=0;
+
 	mutex_lock(&data->config_lock);
-	ret = i2c_smbus_read_byte_data(data->client, command);
+	for(cnt = 0; cnt < 5; cnt++)
+	{
+		ret = i2c_smbus_read_byte_data(data->client, command);
+		if(ret >= 0)
+		{
+			break;
+		}
+	}
 	mutex_unlock(&data->config_lock);
 	if (ret < 0)
 	{
@@ -254,8 +263,17 @@ static s32 mpq8785_write_byte(struct MPQ8785_DRIVER_DATA *data, u8 command,
 							  u8 val)
 {
 	int ret = 0;
+	int cnt=0;
+
 	mutex_lock(&data->config_lock);
-	ret = i2c_smbus_write_byte_data(data->client, command, val);
+	for(cnt = 0; cnt < 5; cnt++)
+	{
+		ret = i2c_smbus_write_byte_data(data->client, command, val);
+		if(ret >= 0)
+		{
+			break;
+		}
+	}
 	mutex_unlock(&data->config_lock);
 	if (ret < 0)
 	{
@@ -285,8 +303,17 @@ static s32 mpq8785_update_byte(struct MPQ8785_DRIVER_DATA *data, u8 command,
 static u16 mpq8785_read_word(struct MPQ8785_DRIVER_DATA *data, u8 command)
 {
 	int ret = 0;
+	int cnt=0;
+
 	mutex_lock(&data->config_lock);
-	ret = i2c_smbus_read_word_data(data->client, command);
+	for(cnt = 0; cnt < 5; cnt++)
+	{
+		ret = i2c_smbus_read_word_data(data->client, command);
+		if(ret >= 0)
+		{
+			break;
+		}
+	}
 	mutex_unlock(&data->config_lock);
 	if (ret < 0)
 	{
@@ -308,8 +335,17 @@ static s32 mpq8785_write_word(struct MPQ8785_DRIVER_DATA *data, u8 command,
 							  u16 val)
 {
 	int ret = 0;
+	int cnt=0;
+
 	mutex_lock(&data->config_lock);
-	ret = i2c_smbus_write_word_data(data->client, command, val);
+	for(cnt = 0; cnt < 5; cnt++)
+	{
+		ret = i2c_smbus_write_word_data(data->client, command, val);
+		if(ret >= 0)
+		{
+			break;
+		}
+	}
 	mutex_unlock(&data->config_lock);
 	if (ret < 0)
 	{
@@ -617,6 +653,52 @@ static int mpq8785_read_string(struct device *dev,
 	return 0;
 }
 
+static int mpq8785_enable(struct device *dev,struct MPQ8785_DRIVER_DATA *data, u8 val)
+{
+	u32 set_value = 0;
+	u32 get_value = 0;
+	int count = 0;
+	int ret;
+
+	if (MPQ8785_MASK_OPERATION_ENABLE == val) {
+		do {
+			ret = mpq8785_update_byte(
+				data, MPQ8785_CMD_OPERATION,
+				MPQ8785_MASK_OPERATION_ENABLE,
+				MPQ8785_MASK_OPERATION_ENABLE);
+			if (ret < 0) {
+				dev_err(dev,"failed to enable output, ret %d\n",ret);
+				return ret;
+			}
+			/* regulator need some time to set up stable output*/
+			msleep(10);
+
+			/*
+		 * The reason for this checking is that,
+		 * it was found that repeatedly toggling the MPQ8785 output voltage
+		 * may cause the voltage to suddenly drop to 0.7V.
+		 * Enabling it a second time may ensures normal output voltage.
+		 */
+			set_value = mpq8785_get_vout(data);
+			get_value = mpq8785_reg2volt(
+				mpq8785_read_word(data, MPQ8785_CMD_READ_VOUT),
+				data->volt_numerator);
+			if (abs((s32)set_value - (s32)get_value) < 20) {
+				return 0;
+			}
+			count++;
+			dev_dbg(dev,
+				"set value(%d )not equal to get value (%d), retry count %d\n",
+				set_value, get_value, count);
+		} while (count < 5);
+		dev_err(dev, "failed to enable output, set value %d, get value %d\n", set_value, get_value);
+		return -EIO;
+	} else {
+		mpq8785_update_byte(data, MPQ8785_CMD_OPERATION,
+				    MPQ8785_MASK_OPERATION_ENABLE, val);
+	}
+	return 0;
+}
 static int mpq8785_write(struct device *dev, enum hwmon_sensor_types type,
 						 u32 attr, int channel, long val)
 {
@@ -631,8 +713,7 @@ static int mpq8785_write(struct device *dev, enum hwmon_sensor_types type,
 		switch (attr)
 		{
 		case hwmon_in_enable:
-			mpq8785_update_byte(data, MPQ8785_CMD_OPERATION,
-								MPQ8785_MASK_OPERATION_ENABLE, (u8)(val << 7));
+			mpq8785_enable(dev,data, (u8)(val << 7));
 			break;
 		case hwmon_in_min:
 			new_value = mpq8785_volt2reg(val, data->volt_numerator);
@@ -992,35 +1073,7 @@ int mpq8785_regulator_enable(struct regulator_dev *rdev)
 	int ret;
 
 	dev_dbg(&rdev->dev, "%s.%d\n", __FUNCTION__, __LINE__);
-
-	do {
-		ret = mpq8785_update_byte(data, MPQ8785_CMD_OPERATION,
-						  MPQ8785_MASK_OPERATION_ENABLE,
-						  MPQ8785_MASK_OPERATION_ENABLE);
-		if (ret < 0) {
-			dev_err(&rdev->dev, "failed to enable output, ret %d\n", ret);
-			return ret;
-		}
-		/* regulator need some time to set up stable output*/
-		msleep(10);
-
-		/*
-		 * The reason for this checking is that,
-		 * it was found that repeatedly toggling the MPQ8785 output voltage
-		 * may cause the voltage to suddenly drop to 0.7V.
-		 * Enabling it a second time may ensures normal output voltage.
-		 */
-		set_value = mpq8785_get_vout(data);
-		get_value= mpq8785_reg2volt(mpq8785_read_word(data, MPQ8785_CMD_READ_VOUT), data->volt_numerator);
-		if (abs((s32)set_value -(s32)get_value) < 20) {
-			return 0;
-		}
-		count++;
-		dev_info(&rdev->dev, "set value(%d )not equal to get value (%d), retry count %d\n", set_value, get_value, count);
-	} while(count < 5);
-
-	dev_err(&rdev->dev, "failed to enable output, set value %d, get value %d\n", set_value, get_value);
-	return -EIO;
+	return mpq8785_enable(&rdev->dev,data,  MPQ8785_MASK_OPERATION_ENABLE);
 }
 
 int mpq8785_regulator_disable(struct regulator_dev *rdev)
-- 
2.50.1

