From 036c10ffeaf96e2edaaa26060e2eb618c5f1201d Mon Sep 17 00:00:00 2001
From: yangwei1 <yangwei1@eswincomputing.com>
Date: Thu, 1 May 2025 18:33:11 +0800
Subject: [PATCH 493/652] fix:clear ecc interrupt anytime

Changelogs:

Signed-off-by: yangwei1 <yangwei1@eswincomputing.com>
---
 drivers/edac/eswin_edac.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/edac/eswin_edac.c b/drivers/edac/eswin_edac.c
index 508010cf4434..ebf42694d3c0 100644
--- a/drivers/edac/eswin_edac.c
+++ b/drivers/edac/eswin_edac.c
@@ -519,8 +519,8 @@ static irqreturn_t intr_handler(int irq, void *dev_id)
 	}
 
 	status = p_data->get_error_info(priv);
-	if (status)
-		return IRQ_NONE;
+	/* if (status)
+		return IRQ_NONE; */
 
 	priv->ce_cnt += priv->stat.ce_cnt;
 	priv->ue_cnt += priv->stat.ue_cnt;
-- 
2.54.0

